### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by mjadach-iv**

### CODE REVIEW

File: src/api/aliases/getAliases.ts [hunk 1]
```diff
@@ -40,14 +40,13 @@ export const getAliases = async (
   const jsonResponse = await rawResponse.json();
   const parsedRes = GetAliasesResponse.safeParse(jsonResponse);
 
-  if(rawResponse.status >= 200 && rawResponse.status < 300) {
-    if (parsedRes.success) {
-      return parsedRes.data;
-    }
+  // parsedRes and error {} from HOPRd have the same type,
+  // we can only rely on rawResponse.ok to know if its a success
+  if (parsedRes.success && rawResponse.ok) {
```
Comment (esterlus) [inline src/api/aliases/getAliases.ts:45]: What about:

```
if (rawResponse.ok) {
    const parsedRes = GetAliasesResponse.safeParse(jsonResponse);
    if (parsedRes.success) {
        return parsedRes.data;
    }
    throw new ZodError(parsedRes.error.issues);
}

const isApiErrorResponse = ApiErrorResponse.safeParse(jsonResponse);

  if (isApiErrorResponse.success) {
    throw new sdkApiError({
      status: rawResponse.status,
      statusText: rawResponse.statusText,
      hoprdErrorPayload: isApiErrorResponse.data
    });
  }

  // we could not parse the response and it is not unexpected
  throw new ZodError(isApiErrorResponse.error.issues);
```

Wouldn't that solve our issues?

File: src/utils/sdkApiError.ts [hunk 2]
```diff
@@ -46,7 +45,12 @@ export class sdkApiError extends Error {
     this.status = customError.status;
     this.statusText = customError.statusText;
     this.hoprdErrorPayload = customError.hoprdErrorPayload;
-    this.description = customError?.hoprdErrorPayload?.error ? customError.hoprdErrorPayload.error : `HTTP Status code ${customError.status}`;
+    this.description = `HTTP Status code ${customError.status}`;
+    if(customError.hoprdErrorPayload?.status) this.description = customError.hoprdErrorPayload.status;
```
Comment (esterlus) [inline src/utils/sdkApiError.ts:49]: Looks to me as if linting is disabled??

**Review by esterlus — CHANGES_REQUESTED** [review]

File: src/api/aliases/getAliases.ts (hunk 1 above)
Comment (mjadach-iv) [reply src/api/aliases/getAliases.ts]: fixed

File: src/utils/sdkApiError.ts (hunk 2 above)
Comment (mjadach-iv) [reply src/utils/sdkApiError.ts]: The rules might not be the best:
image

Comment (mjadach-iv) [reply src/utils/sdkApiError.ts]: Will enable it with full set of linting next
https://github.com/hoprnet/hopr-sdk/issues/126

**Review by esterlus — APPROVED** [review]

**Pull request merged**